### DISCUSSION TITLE
[Feature Request]: Easily configurable timeout periods for ngnix

### DISCUSSION THREAD

**Discussion by MouhcineMAHFOUD** [beyondcode/herd-community#356] [Ideas]
### Feature Description

For the time being, when starting a debugging session, the default timeout of nginx in not enough, it shows a timeout after about 60 seconds
<img width="347" alt="image" src="https://github.com/beyondcode/herd-community/assets/12215673/77e3c537-ca2e-4f48-9868-884f4db26b63">
While the still debugging.

It would be very nice to see such a common configuration easily accessible from Herd's awesome user interface!

Thank you in advance for the efforts!

### Is this feature valuable for other users as well and why?

For me and my team, debugging is a very common thing to do, we always change the default timeouts, this will make a big difference for everyone starting with Herd to have common configurations easily accessible from the gui.

**Comment by aaronaccessvr** [comment]
:+1: For this. I actually had to switch back to Valet because I couldn't figure out how to modify `fastcgi_read_timeout` in Herd. While I think it's completely reasonable to default to a 60 second timeout in 99% of use cases, our use case is a long-running import operation that involves downloading from S3 and uploading to a local MinIO installation. I will probably switch to a Command-based version of this so that I can move back to Herd—Herd is that much more convenient than Valet that it's worth adapting my otherwise working solution. :)

**Comment by mpociot** [comment]
@aaronaccessvr thank you for your feedback. 
How do you modify this via Valet?
All nginx config files are accessible in Herd - so you could just configure it manually 🤔

**Comment by MouhcineMAHFOUD** [comment]
@aaronaccessvr  for my case, I changed the nginx settings in:
"/Users/your-username/Library/Application Support/Herd/config/nginx/nginx.conf"

I added the following section:

```
    sendfile on;
    keepalive_timeout  1d;
    types_hash_max_size 2048;

    send_timeout 1d;
    client_body_timeout 1d;
    client_header_timeout 1d;
    proxy_connect_timeout 1d;
    proxy_read_timeout 1d;
    proxy_send_timeout 1d;
    fastcgi_connect_timeout 1d;
    fastcgi_read_timeout 1d;
    fastcgi_send_timeout 1d;
```

You won't need to do all of them, but I think only updating this one would do: `keepalive_timeout  1d;`

Needless to say that the value '1d' is way exaggerated, so adapt accordingly to your case.

I fact this was very easy to do in Herd as all config was well grouped and accessible. but it would be way better to do it through GUI :D

**Comment by aaronaccessvr** [comment]
Ah! I get three demerit points for not understanding how/where to edit the config files that Herd creates. I was in the Package Contents of the app, not in Application Support. This makes complete sense to me and is a satisfactory solution. Of course, having it in the GUI would be awesome and easier for a lot of other people to get at, but this solution works for me! Yay!

**Comment by aaronaccessvr** [comment]
Maybe a good solution would just be to note where to find the config files in the documentation, potentially in Installation? https://herd.laravel.com/docs/1/getting-started/installation I say this, but perhaps it's just a known standard that such files should be found in Application Support.

**Comment by mpociot** [comment]
We have now documented the config file locations: https://herd.laravel.com/docs/1/advanced-usage/directories-and-files

**Comment by aaronaccessvr** [comment]
Amazing!

*Aaron Collegeman*
Chief Technology Officer
AccessVR
***@***.***
(540) 335-9013
accessvr.com <https://www.accessvr.com>



"Don’t find customers for your product; find products for your customers."
— Seth Godin
